src/java.base/share/classes/java/time/Clock.java

```diff
@@ -759,7 +759,7 @@ public Clock withZone(ZoneId zone) {
         @Override
         public long millis() {
             long millis = baseClock.millis();
-            return millis - Math.floorMod(millis, tickNanos / 1000_000L);
+            return tickNanos < 1000_000L ? millis : millis - Math.floorMod(millis, tickNanos / 1000_000L);
```

Comment:
Cleaner precedence:

```suggestion
            long trunc = (tickNanos >= 1000_000L) ? Math.floorMod(millis, tickNanos / 1000_000L) : 0;
            return millis - trunc;
```

Reply:
Either is fine, the current version avoids any extra operations if no (further) truncation is needed.